### PR TITLE
Add highway recipe

### PR DESCRIPTION
**Pull request by GIGte**
Specify library name and version:  **highway/0.11.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/highway/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "Highway"
+        self.cpp_info.names["cmake_find_package_multi"] = "Highway"
+        self.cpp_info.libs = tools.collect_libs(self)
```
Comment (SpaceIm) [inline recipes/highway/all/conanfile.py:79]:
```suggestion
        self.cpp_info.names["pkg_config"] = "libhwy"
        self.cpp_info.libs = ["hwy"]
```

Upstream doesn't provide an official CMake config file, only a pkg-config file.

File: recipes/highway/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                "{} requires a {} version >= {}"
+                .format(self.name, self.settings.compiler,
+                        self.settings.compiler.version))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
```
Comment (SpaceIm) [inline recipes/highway/all/conanfile.py:55]:
```suggestion

    def _patch_sources(self):
        for patch in self.conan_data["patches"][self.version]:
            tools.patch(**patch)
        # Honor fPIC option
        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
                              "set_property(TARGET hwy PROPERTY POSITION_INDEPENDENT_CODE ON)",
                              "")

```

File: recipes/highway/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTING"] = False
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
```
Comment (SpaceIm) [inline recipes/highway/all/conanfile.py:66]:
```suggestion
        self._patch_sources()
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/highway/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "8",
+            "clang": "7",
+        }
```
Comment (SpaceIm) [inline recipes/highway/all/conanfile.py:35]: They claim to support older compilers. Can't you build hwy target only?

File: recipes/highway/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    name = "highway"
+    description = "Performance-portable, length-agnostic SIMD with runtime " \
+                  "dispatch"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/google/highway"
+    topics = ("highway", "simd")
+
+    settings = "os", "compiler", "build_type", "arch"
+
```
Comment (SpaceIm) [inline recipes/highway/all/conanfile.py:16]:
```suggestion
    settings = "os", "compiler", "build_type", "arch"
    options = {fPIC: [True, False]}
    default_options = {fPIC: True}

```

File: recipes/highway/all/test_package/test_package.cpp [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    const HWY_FULL(float) d;
+    const auto a = Iota(d, 2);
+    const auto b = Iota(d, 3);
+    
+    std::cout << "(0 + 1) + (0 + 1 + 2) = "
+        << GetLane(SumOfLanes(a + b)) << '\n';
+}
+
+} // namespace HWY_NAMESPACE
+HWY_AFTER_NAMESPACE();
```
Comment (SpaceIm) [inline recipes/highway/all/test_package/test_package.cpp:19]: Are you sure it allows to test link to the static lib?

Comment (GIGte) [reply recipes/highway/all/test_package/test_package.cpp:19]: Well, it looks like it doesn't. I will change the test.

File: recipes/highway/all/conanfile.py (hunk 4 above)
Comment (GIGte) [reply recipes/highway/all/conanfile.py:37]: Conan build bot failed on the older compilers (see https://github.com/conan-io/conan-center-index/pull/4996#issuecomment-803603675). And there were errors in the core files, so it won't help if I remove everything else.

Also, they mention testing on specific versions of compilers. On these versions the build succeeds.

>periodically tested on x86 with Clang 7-11 and GCC 8, 9 and 10.2.1

File: recipes/highway/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        # Honor fPIC option
+        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        tools.replace_in_file(cmakelists, "set(CMAKE_POSITION_INDEPENDENT_CODE TRUE), "")
```
[the remark this answers is not in the document]
Comment (GIGte) [reply recipes/highway/all/conanfile.py]: Yeah, I fixed it.

File: recipes/highway/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    name = "highway"
+    description = "Performance-portable, length-agnostic SIMD with runtime " \
+                  "dispatch"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/google/highway"
+    topics = ("highway", "simd")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fPIC": [True, False]}
```
Comment (uilianries) [inline recipes/highway/all/conanfile.py:16]: Do you know why they don't offer shared? Just curious, I didn't find any reference there

Comment (GIGte) [reply recipes/highway/all/conanfile.py:16]: Well, this library is almost header-only. There are only a few functions that are compiled. This is the reason, I think. Also, SIMD libraries are usually all header-only.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**